server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -456,15 +459,21 @@ private static void ensureToBeDecommissionedAttributeWeighedAway(ClusterState st
 
     private static void ensureEligibleRequest(
         DecommissionAttributeMetadata decommissionAttributeMetadata,
-        DecommissionAttribute requestedDecommissionAttribute
+        DecommissionRequest decommissionRequest
     ) {
         String msg = null;
+        DecommissionAttribute requestedDecommissionAttribute = decommissionRequest.getDecommissionAttribute();
         if (decommissionAttributeMetadata != null) {
             // check if the same attribute is registered and handle it accordingly
             if (decommissionAttributeMetadata.decommissionAttribute().equals(requestedDecommissionAttribute)) {
                 switch (decommissionAttributeMetadata.status()) {
-                    // for INIT and FAILED - we are good to process it again
+                    // for INIT - check if it is eligible internal retry
                     case INIT:
+                        msg = (decommissionRequest.originalRequest() == false)
+                            ? "concurrent request received to decommission attribute"
```

Comment:
Might not exactly be concurrent at INIT, make this message to be like illegal state

Reply:
Lets throw `DecommissioningFailedException` independently relying on msg might make it weak
```
if (msg != null) {
            throw new DecommissioningFailedException(requestedDecommissionAttribute, msg);
        }
```

Reply:
Done